packages/embark/src/lib/core/config.js

```diff
@@ -217,60 +227,96 @@ Config.prototype._updateBlockchainCors = function(){
   }
 };
 
-Config.prototype._mergeConfig = function(configFilePath, defaultConfig, env, enabledByDefault) {
+Config.prototype._loadConfigFile = function (configFilePath, defaultConfig, enabledByDefault) {
   if (!configFilePath) {
-    let configToReturn = defaultConfig['default'] || {};
+    const configToReturn = defaultConfig['default'] || {};
     configToReturn.enabled = enabledByDefault || false;
     return configToReturn;
   }
-
-  // due to embark.json; TODO: refactor this
   configFilePath = configFilePath.replace('.json','').replace('.js', '');
-  if (!fs.existsSync(configFilePath + '.js') && !fs.existsSync(configFilePath + '.json')) {
-    // TODO: remove this if
-    if (this.logger) {
-      this.logger.warn(__("no config file found at %s using default config", configFilePath));
-    }
-    return defaultConfig['default'] || {};
-  }
-
   let config;
   if (fs.existsSync(configFilePath + '.js')) {
     delete require.cache[configFilePath + '.js'];
     config = require(configFilePath + '.js');
-  } else {
+  } else if (fs.existsSync(configFilePath + '.json')) {
     config = fs.readJSONSync(configFilePath + '.json');
+  } else {
+    this.logger.warn(__("no config file found at %s using default config", configFilePath));
+    return defaultConfig['default'] || {};
   }
+  return config;
+};
+
+Config.prototype._doMergeConfig = function(config, defaultConfig, env) {
   let configObject = recursiveMerge(defaultConfig, config);
 
   if (env) {
     return recursiveMerge(configObject['default'] || {}, configObject[env]);
+  } else if (env !== false) {
+    this.logger.warn(__("No environment called %s found. Using defaults.", env));
   }
   return configObject;
 };
 
+Config.prototype._mergeConfig = function(configFilePath, defaultConfig, env, enabledByDefault) {
+  const config = this._loadConfigFile(configFilePath, defaultConfig, enabledByDefault);
+  return this._doMergeConfig(config, defaultConfig, env, enabledByDefault);
```

Comment:
So there are a couple of things that I'm wondering here:

1. Is this method ever called as a public API? The `_` should probably indicate that it's a private method. Maybe we shouldn't expose this on the prototype then. It also doesn't seem to have any state, so we can just have a function living inside this module

2. `_mergeConfig()` does more that what it says it does. We're probably better off explicitly loading and then merging where needed. Then we don't have to introduce a thing called `_doMergeConfig()`. If it is indeed necessary to have a method that does more than just merging configs, let's just call what it does... e.g. `loadAndMergeConfig()` or something like that.

Reply:
> Is this method ever called as a public API?

Don't think it is

> let's just call what it does... e.g. loadAndMergeConfig() or something like that.

Good idea. Just for clarity, `_mergeConfig` exsited before, I just extracted part of it.